aiidalab/app.py

```diff
@@ -333,21 +334,47 @@ def find_dependencies_to_install(
             for name, requirement in unmatched_dependencies.items()
         ]
 
-    def _install_dependencies(self, python_bin, stdout=None):
+    def _install_dependencies(self, python_bin, stdout):
         """Try to install the app dependencies with pip (if specified)."""
 
+        def _should_run_reentry():
+            # We do not need to run reentry scan for aiida-core>=2"""
+            packages = find_installed_packages(python_bin=None)
+            installed_aiida = get_package_by_name(packages, "aiida-core")
+            # In practice this should never happen
+            if installed_aiida is None:
+                return False
```

Comment:
In principle, this will never happen but in practice, this is what exactly happened for some leftover versions that didn't specify the version for aiida-core, e.g `aiidalab-widgets-base==1.0.0a1`. Since there is no side effect of running `reentry scan`, can we return True to always running this if not specified?

Reply:
@unkcpz this code path would only happen, if `aiida-core` is not installed. So this doesn't really depend on whether apps depend on `aiida-core` or not. TBH: if `aiida-core` is missing, we will fail anyway later when we try to run restart the daemon...

Reply:
I see, I misunderstood the logic, it was checking with the packages not the requirements of the app. It looks good to me.